connector/dao/src/test/java/it/pagopa/selfcare/mscore/connector/dao/InstitutionConnectorImplTest.java

```diff
@@ -58,6 +58,94 @@ class InstitutionConnectorImplTest {
     @Captor
     ArgumentCaptor<Pageable> pageableArgumentCaptor;
 
+    @Test
+    void testFindAndUpdateInstitutionDataWithNewOnboarding(){
+        DataProtectionOfficer dataProtectionOfficer = new DataProtectionOfficer();
+        dataProtectionOfficer.setAddress("42 Main St");
+        dataProtectionOfficer.setEmail("jane.doe@example.org");
+        dataProtectionOfficer.setPec("Pec");
+
+        PaymentServiceProvider paymentServiceProvider = new PaymentServiceProvider();
+        paymentServiceProvider.setAbiCode("Abi Code");
+        paymentServiceProvider.setBusinessRegisterNumber("42");
+        paymentServiceProvider.setLegalRegisterName("Legal Register Name");
+        paymentServiceProvider.setLegalRegisterNumber("42");
+        paymentServiceProvider.setVatNumberGroup(true);
+
+        InstitutionUpdate institutionUpdate = new InstitutionUpdate();
+        institutionUpdate.setAddress("42 Main St");
+        institutionUpdate.setBusinessRegisterPlace("Business Register Place");
+        institutionUpdate.setDataProtectionOfficer(dataProtectionOfficer);
+        institutionUpdate.setDescription("The characteristics of someone or something");
+        institutionUpdate.setDigitalAddress("42 Main St");
+        institutionUpdate.setGeographicTaxonomies(new ArrayList<>());
+        institutionUpdate.setImported(true);
+        institutionUpdate.setInstitutionType(InstitutionType.PA);
+        institutionUpdate.setPaymentServiceProvider(paymentServiceProvider);
+        institutionUpdate.setRea("Rea");
+        institutionUpdate.setShareCapital("Share Capital");
+        institutionUpdate.setSupportEmail("jane.doe@example.org");
+        institutionUpdate.setSupportPhone("6625550144");
+        institutionUpdate.setTaxCode("Tax Code");
+        institutionUpdate.setZipCode("21654");
+
+        Onboarding onboarding = new Onboarding();
+        onboarding.setBilling(new Billing());
+        onboarding.setContract("contract");
+        onboarding.setStatus(RelationshipState.ACTIVE);
+        onboarding.setCreatedAt(OffsetDateTime.now());
+        onboarding.setProductId("productId");
+        onboarding.setUpdatedAt(OffsetDateTime.now());
+        onboarding.setPricingPlan("pricingPal");
```

Comment:
we could avoid to replicate the same definition of an object for each test method....you might implement an utility test (es. createDummy PaymentServiceProvider) that create a dummy object with default value for each attribute and using that when you need a dummy object for test.....look at https://github.com/pagopa/selfcare-ms-core/blob/main/web/src/test/java/it/pagopa/selfcare/mscore/web/TestUtils.java